### ISSUE TITLE
Push to docker registry fails when using minio azure gateway

### ISSUE THREAD

**Issue by stromvirvel** [minio/minio#8784]
I'm using GitLab on Kubernetes deployed with its official Helm chart. It comes shipped with docker-registry. In addition, I setup minio in gateway mode for Azure Blob Storage. When configuring docker-registry to point to the minio address, pushing docker images is stuck and it retries until eventually a 500 error appears.

## Expected Behavior
It should be able to push the image to Azure without any issues. Also, when running minio in standalone mode, it works fine.

## Current Behavior
```
docker push my.registry.com/repository/image:tag
[ some push and retry output ]
Received unexpected HTTP status: 500 Internal Server Error
```

## Possible Solution
I don't know what the problem is, but #6273 seems a similar issue but with GCS

## Steps to Reproduce (for bugs)
1. setup minio gateway so that it points to Azure Blob storage. Command: `minio gateway azure --compat`
2. setup docker registry so that it points to minio
3. docker push my.registry.com/repository/image:tag

## Context
All access to the Azure Blob storage should be done through minio gateway to simplify data and backup management. I know I can point docker-registry to Azure Blob storage directly, but this has other side effects I'm trying to avoid.

## Regression
No.

## Your Environment
* Version used (`minio version`):
```
/ # minio -v
minio version RELEASE.2020-01-03T19-12-21Z
```
* Environment name and version (e.g. nginx 1.9.1):
```
nginx -v
nginx version: nginx/1.15.5
```
```
/ $ registry -v
registry github.com/docker/distribution v2.7.1
```

* Server type and version:
Official minio docker image.

* Operating System and version (`uname -a`):
Official minio docker image.

* Log docker-registry:
https://gist.github.com/stromvirvel/b5043ca85e023dc88f87f5eca8367601

* Log nginx:
https://gist.github.com/stromvirvel/0795a5bb2136c6cbb0c99603e5e62e1b

**Comment by kannappanr** [comment]
@stromvirvel Can you please run `mc admin trace -v <alias>` against your azure gateway and paste that log also here?

**Comment by stromvirvel** [comment]
@kannappanr 
I started the `trace` command around 11 hours ago, and apparently it just made 4 requests on the very same file, and it's still running:

https://gist.github.com/stromvirvel/d74406ec62768d5133a617dc29a6b41e

Is this expected? Also keep in mind I'm speaking about bucket "docker-registry", the other's aren't affected. Also, ATM the bucket is empty since I've redeployed my app from scratch.
